src/core/data/store.cc

```diff
@@ -163,15 +188,16 @@ void FutureWrapper::initialize_with_identity(int32_t redop_id)
   assert(redop->sizeof_lhs == field_size_);
 #endif
   auto identity = redop->identity;
-  memcpy(ptr, identity, field_size_);
+#ifdef LEGATE_USE_CUDA
+  if (buffer_.get_instance().get_location().kind() == Memory::Kind::GPU_FB_MEM) {
+    auto stream = cuda::StreamPool::get_stream_pool().get_stream();
+    CHECK_CUDA(cudaMemcpyAsync(ptr, identity, field_size_, cudaMemcpyHostToDevice, stream));
```

Comment:
Similar to above, should we deviceSync here?

Reply:
https://github.com/nv-legate/legate.core/pull/353/commits/7f07fa116925cb78585f3ae4b161d64c8b4bb148